### PR TITLE
Add transparant and explicit proxy terraform scripts

### PR DESCRIPTION
**Pull request by abyrne55**
## What does this PR do?
This PR supersedes #267 and implements [OSD-23854](https://issues.redhat.com/browse/OSD-23854), which calls for the code from [abyrne55/transparent-proxy-setup](https://github.com/abyrne55/transparent-proxy-setup) to be moved into this repo's "example" code directory. The two directories created by this PR (vpc-proxied-explicit and vpc-proxied-transparent) contain Terraform scripts for quickly deploying AWS VPCs that explicitly- or transparently-proxy their traffic through a generated proxy server. 

## Checklist
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have tested the functionality against gcp / aws, it doesn't cause any regression~~
- [ ] I have added execution results to the PR's readme

## How to test this PR locally
Please read through and run through the two READMEs included in this PR

### CODE REVIEW

File: examples/aws/terraform/vpc-proxied-explicit/variables.tf [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  description = "Username to use for the proxy machine web UI"
+  type        = string
+  default     = "developer"
+}
+
+# Prefix to add to name tags
+variable "name_prefix" {
+  description = "prefix to add to the Name tag associated with most of the resources created by these scripts"
+  type        = string
+  default     = "transparent-proxy-"
```
Comment (joshbranham) [inline examples/aws/terraform/vpc-proxied-explicit/variables.tf:60]:
```suggestion
  default     = "explicit-proxy-"
```

File: examples/aws/terraform/vpc-proxied-transparent/main.tf [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+  sensitive   = true
+}
+output "proxied_subnet_id" {
+  description = "Proxied subnet ID (launch your test/'captive' instances here)"
+  value       = aws_subnet.proxied.id
+}
+# output "public_subnet_id" {
+#   description = "The ID of the Public Subnet"
+#   value       = aws_subnet.public.id
+# }
```
Comment (joshbranham) [inline examples/aws/terraform/vpc-proxied-transparent/main.tf:195]: Do we need this?

File: examples/aws/terraform/vpc-proxied-explicit/main.tf [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+  value       = "http://${aws_instance.proxy_machine.private_ip}:80"
+}
+output "https_proxy_var" {
+  description = "value for HTTPS_PROXY environmental variable"
+  value       = "https://${aws_instance.proxy_machine.private_ip}:443"
+}
+# output "public_subnet_id" {
+#   description = "The ID of the Public Subnet"
+#   value       = aws_subnet.public.id
+# }
```
Comment (joshbranham) [inline examples/aws/terraform/vpc-proxied-explicit/main.tf:203]: Do we need this?

**Review by joshbranham — COMMENTED** [review]
This looks good to me. I ran a test locally by launching both types of proxy and confirming the proxy was up and running via the web UI. I did not do any further testing after that.

Only other comment I have is we could probably share a lot of this Terraform code in a common module since they are similar stacks, but as far as an example goes this seems fine to me!

/lgtm

File: examples/aws/terraform/vpc-proxied-transparent/main.tf (hunk 2 above)
Comment (abyrne55) [inline examples/aws/terraform/vpc-proxied-transparent/main.tf:195]:
```suggestion
```

Comment (abyrne55) [reply examples/aws/terraform/vpc-proxied-transparent/main.tf:195]: Nope!

File: examples/aws/terraform/vpc-proxied-explicit/main.tf (hunk 3 above)
Comment (abyrne55) [reply examples/aws/terraform/vpc-proxied-explicit/main.tf:203]: Nah

Comment (abyrne55) [inline examples/aws/terraform/vpc-proxied-explicit/main.tf:203]:
```suggestion
```

**Pull request merged**